### PR TITLE
[docs] Add info on how to enable Array syntax usage in Shared routes

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Fixes the issue on how to enable **Array syntax** in an app when using shared routes.

Refs [Slack](https://exponent-internal.slack.com/archives/C1QMWQ1P0/p1723151618462689)

# How

Changes:
- On Shared routes page, add info on how to enable array syntax by using `initialRouteName` in the dynamic layout route file. It continues using the example defined in the Array syntax section. This will act as a source of truth on how to enable this setting.
- On Router Settings page, remove the info that explains the context about using `initialRouteName` for array syntax. This is because the **Array syntax** section can act as a source of truth in itself and we should provide the info on how to enable it within that section.

# Test Plan

Run docs locally and see: http://localhost:3002/router/advanced/shared-routes/#arrays.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by betomoedano — APPROVED** [review]
LGTM

**Review by Simek — APPROVED** [review]
Make sense, thanks! 👍

**Comment by marklawlor** [comment]
Couple of notes that we should probably include.

- You can only provide groups for the current navigator
- If there are two segments groups e.g `/(one)/(two)` only the last segment is used for matching
- If there is no default `initialRouteName`, but there are group `initialRouteNames - the default is the first `group` initialRouteName.

For that last point, let me demostrate:

```
// _layout

export const unstable_settings = {
  "two": {
    initialRouteName: "two"
  }
}

export default function () {
  return (
    <Tabs>
     <Tab.screen name="one" />
     <Tab.screen name="two" />
    </Tab>
  )
}
```

If you link to `/(one)` (a group with no initialRouteName) - it will set the `initialRouteName` as `two`

This is mostly due to a limitation on our integration with React Navigation

File: docs/pages/router/advanced/shared-routes.mdx [hunk 1]
```diff
@@ -43,3 +42,27 @@ export default function DynamicLayout({ segment }) {
...
+export default function DynamicLayout({ segment }) {
+  /* @hide  ... */ /* @end */
+}
+```
+
+In the above example, the `home` route is the default route for the `home` group and the app. The `search` route is the default route for the `search` group.
+
+## Key points
+
+- You can only provide groups for the current navigator.
```
Comment (amandeepmittal) [inline docs/pages/router/advanced/shared-routes.mdx:66]: @marklawlor, can you elaborate more on this point? I'm not sure what "current navigator" means here.

File: docs/pages/router/advanced/shared-routes.mdx [hunk 2]
```diff
@@ -43,3 +42,27 @@ export default function DynamicLayout({ segment }) {
...
+};
+
+export default function DynamicLayout({ segment }) {
+  /* @hide  ... */ /* @end */
+}
+```
+
+In the above example, the `home` route is the default route for the `home` group and the app. The `search` route is the default route for the `search` group.
+
+## Key points
```
Comment (amandeepmittal) [inline docs/pages/router/advanced/shared-routes.mdx:64]: @marklawlor, please check if mentioning the info you shared as "key points" is enough to convey the information.

**Pull request merged**